### PR TITLE
Dev/multimodal format & basic processing pipeline for multimodal datasets support

### PR DESCRIPTION
**Pull request by HYLcool**
- Support the basic Data-Juicer intermediate format for multimodal datasets. The detailed format information will be clarified later in some docs.
  - Support image loading, new configs, new special tokens, new OP abstraction
- Add a basic image OP: image_aspect_ratio_filter. -- for basic pipeline test
- Add two format conversion tools: `llava2jd` and `dj2llava`, which support convert LLaVA-like datasets to Data-Juicer format and reverse. -- for basic pipeline test

### CODE REVIEW

**Comment by HYLcool** [comment]
- [ ] This PR is under an internal test now.

**Review by zhijianma — APPROVED** [review]
LGTM

**Review by yxdyc — COMMENTED** [review]
leaved some minor suggestions

File: data_juicer/config/config.py [hunk 1]
```diff
@@ -396,6 +437,7 @@ def sort_op_by_types_and_names(op_name_classes):
 
 def config_backup(cfg):
     cfg_path = cfg.config[0].absolute
+    cfg.cfg_path = os.path.dirname(cfg_path)  # record the path of config
```
[the remark this answers is not in the document]
Comment (HYLcool) [reply data_juicer/config/config.py:440]: This line of code is removed due to cfg_path/cfg_dir is unused later.

**Review by yxdyc — APPROVED** [review]

**Pull request merged**